File: android/src/main/kotlin/me/yohom/amapbase/navi/AMapNavOptions.java
package me.yohom.amapbase.navi;

import com.amap.api.maps.model.LatLng;

public class AMapNavOptions {

    public static final int NAVI_TYPE_DRIVER = 0;
    public static final int NAVI_TYPE_WALK = 1;
    public static final int NAVI_TYPE_RIDE = 2;

    public AMapNavOptions(int navType, LatLng startLocation, LatLng endLocation, double bottomContentH, boolean useEmulatorNavi) {
        this.navType = navType;
        this.startLocation = startLocation;
        this.endLocation = endLocation;
        this.bottomContentH = bottomContentH;
        this.useEmulatorNavi = useEmulatorNavi;
    }

    /// 导航模式
    int navType;

    /// 导航起点
    LatLng startLocation;

    /// 导航终点
    LatLng endLocation;

    ///底部地图内容绘制区域向上偏移的高度
    double bottomContentH;

    /// 是否使用虚拟导航（模拟导航）
    boolean useEmulatorNavi;

    public int getNavType() {
        return navType;
    }

    public LatLng getStartLocation() {
        return startLocation;
    }

    public LatLng getEndLocation() {
        return endLocation;
    }

    public double getBottomContentH() {
        return bottomContentH;
    }

    public boolean isUseEmulatorNavi() {
        return useEmulatorNavi;
    }
}

File: lib/src/navi/model/amap_nav_options.dart
import 'dart:convert';

import 'package:flutter_amap_base/src/map/model/latlng.dart';

const NAVI_TYPE_DRIVER = 0;
const NAVI_TYPE_WALK = 1;
const NAVI_TYPE_RIDE = 2;

class AMapNavOptions {
  const AMapNavOptions({
    this.navType = NAVI_TYPE_DRIVER,
    required this.startLocation,
    required this.endLocation,
    this.bottomContentH = 100,
    this.useEmulatorNavi = false,
  });

  /// 导航模式
  final int navType;

  /// 导航起点
  final LatLng startLocation;

  /// 导航终点
  final LatLng endLocation;

  ///底部地图内容绘制区域向上偏移的高度
  final double bottomContentH;

  /// 是否使用虚拟导航（模拟导航），false 则为 GPS 真实导航
  final bool useEmulatorNavi;

  Map<String, dynamic> toJson() {
    return {
      'navType': navType,
      'startLocation': startLocation.toJson(),
      'endLocation': endLocation.toJson(),
      'bottomContentH': bottomContentH,
      'useEmulatorNavi': useEmulatorNavi,
    };
  }

  String toJsonString() => jsonEncode(toJson());

  AMapNavOptions copyWith({
    int? navType,
    bool? useEmulatorNavi,
  }) {
    return AMapNavOptions(
      navType: navType ?? this.navType,
      startLocation: startLocation ?? this.startLocation,
      endLocation: endLocation ?? this.endLocation,
      bottomContentH: bottomContentH ?? this.bottomContentH,
      useEmulatorNavi: useEmulatorNavi ?? this.useEmulatorNavi,
    );
  }

  @override
  String toString() {
    return 'AMapNavOptions{navType: $navType, startLocation:$startLocation, endLocation:$endLocation, bottomContentH:$bottomContentH}';
  }
}
